common/src/Sync.php

```diff
@@ -0,0 +1,129 @@
+
+namespace Valued\PrestaShop;
+
+use Configuration;
+use Customer;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use ModuleFrontController;
+use PrestaShop\Module\ProductComment\Entity\ProductComment;
+use PrestaShop\PrestaShop\Adapter\Validate;
+use Product;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class Sync extends ModuleFrontController {
+
+    /** @var bool */
+    public $ajax;
+
+    /**
+     * @throws Exception
+     */
+    public function postProcess() {
+        $request_data = trim(file_get_contents('php://input'));
+        if (!$request_data) {
+            throw new HttpException(400, 'Empty request data');
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            throw new HttpException(400, 'Invalid JSON data provided');
+        }
+
+        if (
+            !$this->hasCredentialFields($request_data['webshop_id'], $request_data['api_key'])
+            || $this->credentialsEmpty($request_data['webshop_id'], $request_data['api_key'])
+        ) {
+            throw new HttpException(403, 'Missing credential fields');
+        }
+
+        $this->isAuthorized($request_data['webshop_id'], $request_data['api_key']);
+
+        $lang_id = (int) Configuration::get('PS_LANG_DEFAULT');
+        $product = new Product($request_data['product_review']['product_id'], false, $lang_id);
+        if (!Validate::isLoadedObject($product)) {
+            throw new HttpException(404, sprintf('Could not find product with ID (%d)', $request_data['product_review']['product_id']));
+        }
+
+        if (!Configuration::get($this->module->getConfigName('SYNC_PROD_REVIEWS'))) {
+            throw new HttpException(403, 'Product review sync is disabled.');
+        }
+
+        $this->syncProductReview($request_data['product_review']);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function syncProductReview(array $product_review) {
+        $this->ajax = 1;
+        /** @var EntityManagerInterface $entityManager */
+        $entity_manager = $this->container->get('doctrine.orm.entity_manager');
+        $product_comment_entity = new ProductComment();
+        $date_add = DateTime::createFromFormat('Y-m-d H:i:s', $product_review['created']);
+
+        if ($product_review['id']) {
+            $product_comment_repository = $entity_manager->getRepository(ProductComment::class);
+            $product_comment = $product_comment_repository->find($product_review['id']);
+            if ($product_review['deleted']) {
+                $product_comment->setDeleted(1);
+                $this->logReviewSync($product_review['id'], true);
+            } else {
+                $product_comment->setDeleted(0);
+                $this->logReviewSync($product_review['id']);
+            }
+            $entity_manager->flush();
+            $this->ajaxRender(json_encode(['review_id' => $product_review['id']], JSON_PARTIAL_OUTPUT_ON_ERROR));
+            die();
```

Comment:
In case the existing review has been changed, e.g. changed text/rating/title, the review will not be updated.